File: backend/mini-xrm/src/main/java/hu/rrsoftware/minixrm/partner/PartnerService.java
package hu.rrsoftware.minixrm.partner;

import hu.rrsoftware.minixrm.enums.QualificationType;
import hu.rrsoftware.minixrm.partner.dto.PartnerDetailDto;
import hu.rrsoftware.minixrm.partner.dto.PartnerRequestDto;
import hu.rrsoftware.minixrm.partner.dto.PartnerResponseDto;

import java.util.List;

public interface PartnerService {

    PartnerResponseDto create(PartnerRequestDto dto);

    PartnerResponseDto update(Long id, PartnerRequestDto dto);

    void delete(Long id);

    PartnerResponseDto getById(Long id);

    PartnerDetailDto getDetailById(Long id);

    List<PartnerResponseDto> getAll(QualificationType qualification);
}

File: frontend/src/app/components/partner/partner-service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {QualificationType} from '../enums/QualificationType';
import {Observable} from 'rxjs';
import {PartnerResponseDto} from './dto/PartnerResponseDto';
import {PartnerRequestDto} from './dto/PartnerRequestDto';

@Injectable({
  providedIn: 'root',
})
export class PartnerService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = 'http://localhost:8080/api/partners';

  getAll(qualifications?: QualificationType[]): Observable<PartnerResponseDto[]> {
    let params = new HttpParams();

    qualifications?.forEach(qualification => {
      params = params.append('qualification', qualification);
    })
    return this.http.get<PartnerResponseDto[]>(this.baseUrl, { params });
  }

  create(request: PartnerRequestDto): Observable<PartnerResponseDto> {
    return this.http.post<PartnerResponseDto>(this.baseUrl, request);
  }
}
